Mathlib/CategoryTheory/Category/Cat/CartesianClosed.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2025 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Emily Riehl
+-/
+import Mathlib.CategoryTheory.Closed.Cartesian
+import Mathlib.CategoryTheory.Functor.Currying
+import Mathlib.CategoryTheory.Monoidal.Cartesian.Cat
+
+/-!
+# Cartesian closed structure on `Cat`
+
+The category of small categories is cartesian closed, with the exponential at a category `C`
+defined by the functor category mapping out of `C`.
+
+Adjoint transposition is defined by currying and uncurrying.
+
+-/
+
+universe v u v₁ v₂ v₃ v₄ u₁ u₂ u₃ u₄
+
+namespace CategoryTheory
+
+open Functor
+
+namespace Cat
+
+variable (C : Type u) [Category.{v} C]
+
+/-- A category `C` induces a functor from `Cat` to itself defined
+by forming the category of functors out of `C`. -/
+def exp : Cat ⥤ Cat where
+  obj D := Cat.of (C ⥤ D)
+  map F := {
+    obj G := G ⋙ F
+    map α := whiskerRight α F
+  }
```

Comment:
```suggestion
  map F :=   map F := (whiskeringRight _ _ _).obj F
```

Reply:
No, its a bit confusing, [`whiskeringRight`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/CategoryTheory/Whiskering.html#CategoryTheory.whiskeringRight) is the functor that takes a functor `F` and returns the functor (between functor categories) given by postcomposing with `F` (so `(whiskeringRight _ _ _).obj F` is by definition what you wrote).